pkg/validator/validators.go

```diff
@@ -330,12 +341,32 @@ func (kc *validator) validateMRCOnUpdate(oldMRC *configv1alpha2.MeshRootCertific
 		return fmt.Errorf("cannot update SpiffeEnabled for MRC %s. Create a new MRC and initiate root certificate rotation to enable SPIFFE certificates", getNamespacedMRC(oldMRC))
 	}
 
-	// TODO(#5205): add validation for simplified root certificate rotation process
+	// check for intent changes that are always invalid
+	if oldMRC.Spec.Intent == configv1alpha2.ActiveIntent && newMRC.Spec.Intent == configv1alpha2.InactiveIntent {
+		return fmt.Errorf("cannot move an Active to Passive for MRC %s", getNamespacedMRC(oldMRC))
+	}
+
+	if oldMRC.Spec.Intent == configv1alpha2.InactiveIntent && newMRC.Spec.Intent == configv1alpha2.ActiveIntent {
+		return fmt.Errorf("cannot move an Active to Passive for MRC %s", getNamespacedMRC(oldMRC))
+	}
+
+	// does this change create a valid configuration?
+	// at this point we don't care about order just if combination is valid
+	err := kc.simulateDesiredCertConfiguration(newMRC)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
 func validateMRCProvider(mrc *configv1alpha2.MeshRootCertificate) error {
 	p := mrc.Spec.Provider
+
+	if p.Tresor == nil && p.CertManager == nil && p.Vault == nil {
```

Comment:
This scenario should be handled by the logical constraints in the CRD schema. I don't think it hurts to validate it here again though.